README.rst: Do minor cleanup and add self as a maintainer (#434)
cpplint/cpplint: README.rst

Cpplint is a command-line tool to check C/C++ files for style issues according to `Google's C++ style guide <http://google.github.io/styleguide/cppguide.html>`_.

> Cpplint is a friendly fork of Google's original tool at `google/styleguide <https://github.com/google/styleguide>`_. Nowadays, `Google is no longer maintaining their public version of cpplint <https://github.com/google/styleguide/pull/528#issuecomment-592315430>`_, and many issues and pull requests remain unimplemented.

Reviewer: ```suggestion
Cpplint is a friendly fork of Google's original tool at `google/styleguide <https://github.com/google/styleguide>`_. Nowadays, `Google no longer maintains a public version of cpplint <https://github.com/google/styleguide/pull/528#issuecomment-592315430>`_, and many issues and pull requests remain unimplemented.
```
"their public version" implies they have a public version

Author: They ___had___ a public version at https://github.com/google/cpplint but that is gone.  They may ___still___ have a private, internal version but there is no way to know.

___This___ is a public version of cpplint, which is not Google's version cpplint.

Reviewer: Yes, exactly. "Their" implies they still have a separate public version when it has been removed. As you've mentioned, this is a public version, hence the change to "a public version".

> Cpplint is a friendly fork of Google's original tool at `google/styleguide <https://github.com/google/styleguide>`_. Nowadays, `Google is no longer maintaining their public version of cpplint <https://github.com/google/styleguide/pull/528#issuecomment-592315430>`_, and many issues and pull requests remain unimplemented.

Reviewer: we should look at those pull requests ;) /t

Author: I do not know how we can see them anymore.

Reviewer: I'm teasing about the one I've made lol

This fork aims to update cpplint to modern specifications, and be (somewhat) more open to adding fixes and features to make cpplint usable in wider contexts.

Installation
============

> Use `pipx <https://pipx.pypa.io>`_ to install cpplint from PyPI, run:
> 
> .. code-block:: bash
> 
>     $ pipx install cpplint
> 
> Or use `uv <https://docs.astral.sh/uv>`_ to install cpplint from PyPI, run:

Reviewer: ```suggestion
To use `uv <https://docs.astral.sh/uv>`_ to install cpplint from PyPI as a tool, run:
```

Methinks pipx is a little outmoded

Since it's just a PyPI package name, I think just doing one PyPI-compliant tool is enough unless it can trigger some special hijinks. I don't want this to attract flies that try to add every PyPI tool in existence.

Author: `pipx` comes from the Python Packaging Authority, like `pip` does, so some people may trust it more.

`uv` was recently [purchased by OpenAI](https://astral.sh/blog/openai), which some people might not trust in their daily toolchain.

Reviewer: I think we can trust users to select the tool that they'd like, and it's clear enough that `cpplint` is the package name on PyPI to feed into the tool that they'd like.

Author: Why are you blocking progress over this?

Reviewer: i'm not; this is the smallest thing among all my comments here. I won't block because of this as long as 1. we go "to use" instead of "use" for grammar 2. the other things above are resolved.

TL;DR: I'm blocking because of the version number–bumping precedent this is setting for the hook, not this.

Author: Use this or use that proper English.

Reviewer: ("Use" itself isn't the grammar problem, it's the comma. You've got two independent clauses joined with a comma splice. "To" makes the first one dependent and fix the grammar error.)

Again, though, I think the above thread's far more important.

Cpplint is a command-line tool to check C/C++ files for style issues according to `Google's C++ style guide <http://google.github.io/styleguide/cppguide.html>`_.

> Cpplint is a friendly fork of the original tool that was developed and maintained by Google Inc. at `google/styleguide <https://github.com/google/styleguide>`_. Nowadays, `Google is no longer maintaining their public version of cpplint <https://github.com/google/styleguide/pull/528#issuecomment-592315430>`_, and pretty much everything in their repo's PRs and issues about cpplint have gone unimplemented.

Author: Nice!  Fixed.

    $ pipx install cpplint

Or use [`uv`](https://docs.astral.sh/uv) to install cpplint from PyPI, run:

.. code-block:: bash

    $ uv tool install cpplint

> 

Author: Nice!  Fixed.

.. code-block:: yaml

  - repo: https://github.com/cpplint/cpplint

>     rev: 2.0.2

Reviewer: I don't like the implication that we'll have to update a version number in the README every release. Maybe we could pin this to a version number significant to us, or just make it like this:?
```suggestion
    rev:  # cpplint version number
```

Author: Nope.  That is not usable out-of-the-box.

Reviewer: I don't see why it has to be. It's very clear how it can be usable out-of-the-box. READMEs are usually quite static.

Author: Let's agree to disagree.  Repos occasionally do `pre-commit autoupdate` but I am OK for them to use an older version until they do so.

Should we add a step in our release process to update the README?

Reviewer: > I am OK for them to use an older version until they do so.

Alright, let's

> pin this to a version number significant to us

, then. I personally want to pin it at the next release <small>(I'm thinking of making it 2.1.0. We should be able to to do the release process after the currently open PRs are merged, which shouldn't take too long and we can wait for. Even #426 where the PR creator hasn't responded I should be able to fix up quickly)</small> because of how wide the namespace indentation false positives it fixes are.

I do not want to bump it every release, as you suggest, though. That clogs up the README's file history without introducing new information.

Changes
=======

* Python 3 compatibility
* JUnit XML output format
* More default file extensions
* Continuous integration on GitHub
* Support for excluding files via ``--exclude``
* Customizable file extensions with the ``--extensions`` argument
* Support for recursive file discovery via the ``--recursive`` argument

> * Overriding repository root auto-detection via ``--repository``

Reviewer: Nice. I don't feel like `--repository` is nearly as important compared to the others, though. If we're at this, maybe we could highlight our NOLINT blocks and commas; or support for modern language updates (e.g. recognizing C++20 headers, attributes)?

Author: Separate PR.
